testcharms/charm.go

```diff
@@ -163,3 +167,39 @@ func SetPublicWithResources(c *gc.C, client *csclient.Client, id *charm.URL, res
 func SetPublic(c *gc.C, client *csclient.Client, id *charm.URL) {
 	SetPublicWithResources(c, client, id, nil)
 }
+
+// CheckCharmReady ensures that a desired charm archive exists and
+// has some content.
+func CheckCharmReady(c *gc.C, charmArchive *charm.CharmArchive) {
+	fileSize := func() int64 {
+		f, err := os.Open(charmArchive.Path)
+		c.Assert(err, jc.ErrorIsNil)
+		defer f.Close()
+
+		fi, err := f.Stat()
+		c.Assert(err, jc.ErrorIsNil)
+		return fi.Size()
+	}
+
+	oldSize := int64(0)
+	var currentSize int64
+	charmReady := func() bool {
```

Comment:
I have considered it originally. However, it does mean an additional pass through the loop.

With current implementation, the loop is essentially while-do and would give you just one pass:
```
1: new file size 1940 (old size 0)
```
With the logic within the loop, we'd have do-while loop, with the output of:
```
1: new file size 1940 (old size 0)
2: new file size 1940 (old size 1940)
```
However, since we have both instinctively went to logic-within-the-loop, I'll revert my implementation to it.